src/test/java/de/cronn/validation_files_diff/ValidationDiffTest.java

```diff
@@ -154,30 +151,27 @@ public void testGetDirDiffElementFromPath() {
 	}
 
 	private void setupGeneralMocks() {
-		project = mock(Project.class);
-		disposable = mock(Disposable.class);
 		virtualFile = mock(VirtualFile.class);
-		moduleManager = mock(ModuleManager.class);
 		dirDiffManager = mock(DirDiffManager.class);
+		VirtualFileManager virtualFileManager = mock(VirtualFileManager.class);
 		fileSystem = mock(LocalFileSystem.class);
 		moduleAnalyser = mock(ModuleAnalyser.class);
 
-		projectSettings = new ValidationDiffProjectOptionsProviderImpl();
+		disposable = Disposer.newDisposable();
+		MockApplication application = MockApplication.setUp(disposable);
+		project = new MockProject(application.getPicoContainer(), disposable);
+
 		applicationSettings = new ValidationDiffApplicationOptionsProviderImpl();
+		application.registerService(ValidationDiffApplicationOptionsProvider.class, applicationSettings);
 
-		PowerMockito.mockStatic(ModuleManager.class);
-		PowerMockito.mockStatic(ValidationDiffProjectOptionsProvider.class);
-		PowerMockito.mockStatic(ValidationDiffApplicationOptionsProvider.class);
-		PowerMockito.mockStatic(DirDiffManager.class);
-		PowerMockito.mockStatic(LocalFileSystem.class);
+		doReturn(fileSystem).when(virtualFileManager).getFileSystem(LocalFileSystem.PROTOCOL);
+		application.registerService(VirtualFileManager.class, virtualFileManager);
 
-		PowerMockito.when(ModuleManager.getInstance(any())).thenAnswer(invocation -> moduleManager);
-		PowerMockito.when(ValidationDiffApplicationOptionsProvider.getInstance()).thenAnswer(invocation -> applicationSettings);
-		PowerMockito.when(ValidationDiffProjectOptionsProvider.getInstance(eq(project))).thenAnswer(invocation -> projectSettings);
-		PowerMockito.when(DirDiffManager.getInstance(eq(project))).thenAnswer(invocation -> dirDiffManager);
-		PowerMockito.when(LocalFileSystem.getInstance()).thenAnswer(invocation -> fileSystem);
+		project.registerService(ValidationDiffProjectOptionsProvider.class, new ValidationDiffProjectOptionsProviderImpl());
+		project.registerService(DirDiffManager.class, dirDiffManager);
```

Comment:
Nice to see that getting rid of the static mocking brings you closer to the real setup. :+1:

Reply:
:)